tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -37,6 +43,19 @@ using namespace llvm;
 
 using namespace sourcekitd_test;
 
+#if defined(_WIN32)
+namespace {
+int STDOUT_FILENO = _fileno(stdout);
+const constexpr size_t MAXPATHLEN = MAX_PATH + 1;
+char *realpath(const char *path, const char *resolved_path) {
```

Comment:
It's plausible there's already an API in `llvm/FileSystem.h` or 'llvm/Path.h` that does this in a platform-independent way, could you check ?